app/web/service/notifications.ts

```diff
@@ -65,3 +68,21 @@ export async function registerPushNotificationSubscription(
 export async function sendTestPushNotification() {
   await client.notifications.sendTestPushNotification(new Empty());
 }
+
+export async function listNotifications() {
+  const req = new ListNotificationsReq();
+  const res = await client.notifications.listNotifications(req);
+  return res.toObject();
+}
+
+export async function markAllNotificationsSeen() {
+  const req = new MarkAllNotificationsSeenReq();
```

Comment:
You need to pass in the latest notification ID you have (so that if the user marks all seen but there was a new one between page load and now, they don't miss the new one).